lib/internal/Magento/Framework/MultiDimensionalIndexer/IndexName.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Framework\MultiDimensionalIndexer;
+
+/**
+ * Index Name object
+ *
+ * @api
+ */
+class IndexName
+{
+    /**
+     * @var string
+     */
+    private $indexId;
+
+    /**
+     * @var Dimension[]
+     */
+    private $dimensions;
+
+    /**
+     * @var Alias
+     */
+    private $alias;
+
+    /**
+     * @param string $indexId
+     * @param Dimension[] $dimensions
+     * @param Alias $alias
+     */
+    public function __construct(string $indexId, array $dimensions, Alias $alias)
```

Comment:
To make this code more safe I would suggest checking types inside `$dimensions` array.

Reply:
Done.